### PR TITLE
api,middlware: add Indent middleware

### PR DESCRIPTION
**Pull request by chappjc**

### CODE REVIEW

**Comment by buck54321** [comment]
Is it possible to add the indent middleware to the insight API too?

**Comment by chappjc** [comment]
Yeah, it would work there too.  I wanted to start with these bits since they were recently changed in https://github.com/decred/dcrdata/pull/1698, but now is the time to apply it everywhere.  I think the Insight API also lacks validation of the indent query value too, so that's another good reason to do it.

File: api/insight/apiroutes.go [hunk 1]
```diff
@@ -86,17 +86,16 @@ type InsightApi struct {
 // NewInsightApi is the constructor for InsightApi.
 func NewInsightApi(client *rpcclient.Client, blockData BlockDataSource, params *chaincfg.Params,
 	memPoolData rpcutils.MempoolAddressChecker, JSONIndent string, maxAddrs int, status *apitypes.Status) *InsightApi {
```
Comment (chappjc) [inline api/insight/apiroutes.go]: horrible that the `JSONIndent` input arg was just unused and named.

**Comment by chappjc** [comment]
Finally got around to testing it.

```
$  curl http://127.0.0.1:7777/api/block/434568/verbose?indent=truadsf
Bad Request
$  curl http://127.0.0.1:7777/api/block/434568/verbose?indent=true
{
   "hash": "00000000000000000a8ec60d6c24d311407393a24156f30562559bb66c904c2d",
   "confirmations": 7,
   "size": 42266,
   "height": 434568,
   "version": 7,
   "merkleroot": "9cfb444968c26c35de51f4badfc2597a782ecefdb69173a4916a13850b0b70fb",
   "stakeroot": "f8ce8e8cacbb7da0efa8aa72de29da0004513823ad44b0447b02924fb76a4da6",
   "tx": [
...
$  curl http://127.0.0.1:7777/api/block/434568/verbose?indent=f
{"hash":"00000000000000000a8ec60d6c24d311407393a24156f30562559bb66c904c2d","confirmations":7,"size":42266,"height":434568,"version":7,"merkleroot":"9cfb444968c26c35de51f4badfc2597a782ecefdb69173a4916a13850b0b70fb","stakeroot":"f8ce8e8cacbb7da0efa8aa72de29da0004513823ad44b0447b02924fb76a4da6","tx":[
...
$  curl http://127.0.0.1:7777/api/block/434568/verbose?indent=0
{"hash":"00000000000000000a8ec60d6c24d311407393a24156f30562559bb66c904c2d","confirmations":7,"size":42266,"height":434568,"version":7,"merkleroot":"9cfb444968c26c35de51f4badfc2597a782ecefdb69173a4916a13850b0b70fb","stakeroot":"f8ce8e8cacbb7da0efa8aa72de29da0004513823ad44b0447b02924fb76a4da6","tx":[
...
```

**Pull request merged**